modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -347,35 +375,44 @@
     /** Spring context, potentially {@code null}. */
     private GridSpringResourceContext rsrcCtx;
 
-    /** */
+    /**
+     * The instance of scheduled thread pool starvation checker. {@code null} if starvation checks have been
+     * disabled by the value {@link IgniteSystemProperties#IGNITE_STARVATION_CHECK_INTERVAL} system property.
+     */
     @GridToStringExclude
     private GridTimeoutProcessor.CancelableTask starveTask;
 
-    /** */
+    /**
+     * The instance of scheduled metrics logger. {@code null} means that the metrics logges have been disabled
+     * by configuration. See {@link IgniteConfiguration#getMetricsLogFrequency()} for details.
+     */
     @GridToStringExclude
     private GridTimeoutProcessor.CancelableTask metricsLogTask;
 
-    /** */
+    /**
+     * The instance of scheduled long operation checker. {@code null} means that the operations checker is disabled
+     * by the value of {@link IgniteSystemProperties#IGNITE_LONG_OPERATIONS_DUMP_TIMEOUT} system property.
+     */
     @GridToStringExclude
     private GridTimeoutProcessor.CancelableTask longOpDumpTask;
 
-    /** Indicate error on grid stop. */
+    /** {@code true} if an error occur at Ignite instance stop. */
     @GridToStringExclude
     private boolean errOnStop;
 
-    /** Scheduler. */
+    /** An instance of the scheduler which provides functionality for scheduling jobs locally. */
     @GridToStringExclude
     private IgniteScheduler scheduler;
 
-    /** Kernal gateway. */
+    /** The kernal state guard. See {@link GridKernalGateway} for details. */
     @GridToStringExclude
     private final AtomicReference<GridKernalGateway> gw = new AtomicReference<>();
 
-    /** Stop guard. */
+    /** Flag indicates that the ignite instance is scheduled to be stopped. */
     @GridToStringExclude
     private final AtomicBoolean stopGuard = new AtomicBoolean();
 
-    /** */
+    /** The state object is used to handle if reconnection occurred. See {@link IgniteKernal#onReconnected(boolean)}. */
```

Comment:
Suggestions to rearrange:
* The state object is used _to handle reconnection_.
* The state object is used _when reconnection occurs_.

Reply:
Agree, fixed.